include/vcpkg/base/parallel-algorithms.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+// At this time only MSVC supports parallel algorithms by default
+// Gcc needs tbb as a runtime dependency
+// Clang doesn't support parallel algorithms at all
+
+#if defined(_MSC_VER)
+#include <algorithm>
+#include <execution>
+#include <mutex>
+
+#define vcpkg_parallel_for_each(BEGIN, END, CB) std::for_each(std::execution::par, BEGIN, END, CB)
+#define vcpkg_par_unseq_for_each(BEGIN, END, CB) std::for_each(std::execution::par_unseq, BEGIN, END, CB)
+
+#else
+#include <atomic>
+#include <future>
+#include <vector>
+
+namespace vcpkg
+{
+    template<class It, class F>
+    void parallel_for_each(It begin, It end, F&& cb)
+    {
+        if (begin == end)
+        {
+            return;
+        }
+        if (begin + 1 == end)
+        {
+            return cb(std::move(*begin));
+        }
+
+        const auto thread_count = std::thread::hardware_concurrency() * 2;
```

Comment:
```suggestion
        const auto thread_count = get_concurrency();
```
You then have to include `#include <vcpkg/base/system.h>`

Reply:
I think this doesn't make much sense because this is ignored by `std::for_each` on Windows anyway.

Reply:
Hmm still needs to adjust to `VCPKG_MAX_CONCURRENCY/VCPKG_CONCURRENCY`

Reply:
> Hmm still needs to adjust to `VCPKG_MAX_CONCURRENCY/VCPKG_CONCURRENCY`

`get_concurrency()` does that

Reply:
If we had access to the underlying thread pool we could use [SetThreadpoolThreadMaximum](https://learn.microsoft.com/en-us/windows/win32/api/threadpoolapiset/nf-threadpoolapiset-setthreadpoolthreadmaximum).

Reply:
Or we use the custom implementation also on windows

Reply:
or just use `hwloc` ;) A wait vcpkg is a package manager and cannot use other stuff and has to come up with its own solution...